src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -76,43 +85,27 @@ public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool
             else
             {
                 fileToOpen = GetTestFilePath() + ".txt";
-                File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}");
+                File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled)}");
             }
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || _allowedProgramsToRun.Any(program => IsProgramInstalled(program)))
             {
-                string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
-                foreach (var program in allowedProgramsToRun)
+                using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }))
                 {
-                    if (IsProgramInstalled(program))
+                    Assert.NotNull(px); // on Windows px is null for some reason
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                     {
-                        programToOpen = program;
-                        break;
+                        // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason. Refer to #23972
+                        Console.WriteLine($"{nameof(ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled)}(isFolder: {isFolder}), ProcessName: {px.ProcessName}");
                     }
+                    else
+                    {
+                        Assert.Equal(programToOpen, px.ProcessName);
```

Comment:
This is causing this test to fail on this PR on centos and RH:
https://mc.dot.net/#/user/maryamariyan/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/707d2a7aefcf8d71e5e8d03e299bcdcc289134d2/workItem/System.Diagnostics.Process.Tests

It looks like `programToOpen` is null and `px.ProcessName` is xdg-open.

Reply:
probably because you aren't initializing `programToOpen` anywhere